### ISSUE TITLE
[Bug]: Toolbar button iconography is absent.

### ISSUE THREAD

**Issue by RokeJulianLockhart** [nextcloud/desktop#7419]
### ⚠️ Before submitting, please verify the following: ⚠️

- [x] This is a **bug**, not a question or a configuration issue.
- [x] This issue is **not** already reported on Github (I've searched it).
- [x] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [x] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

As the undermentioned screenshot demonstrates, the toolbar button iconography is absent:

> ![Image](https://github.com/user-attachments/assets/50ef1dca-46c2-4129-9a1c-1fda16a44a59)

### Steps to reproduce

1. Install https://fedora.pkgs.org/rawhide/fedora-x86_64/nextcloud-client-3.14.2-1.fc42.x86_64.rpm.html (as a dependency of https://fedora.pkgs.org/rawhide/fedora-x86_64/nextcloud-client-dolphin-3.14.2-1.fc42.x86_64.rpm.html):

   ```sh
   #!/usr/bin/env -S sh
   sudo dnf install 'nextcloud-client-3.14.2-2.fc40.x86_64'
   ```

3. Invoke the GUI:

   ```sh
   #!/usr/bin/env -S sh
   nextcloud-client
   ```

### Expected behavior

The icons should be rendered.

### Which files are affected by this bug

file:///usr/bin/nextcloud

### Operating system

Linux

### Which version of the operating system you are running.

https://download.fedoraproject.org/pub/fedora/linux/releases/40/Spins/x86_64/iso/Fedora-KDE-Live-x86_64-40-1.14.iso

### Package

Distro package manager

### Nextcloud Server version

[28.0.10.1](https://wim.nl.tab.digital/status.php#:~:text=version%09%2228.0.10.1%22)

### Nextcloud Desktop Client version

https://fedora.pkgs.org/rawhide/fedora-x86_64/nextcloud-client-3.14.2-1.fc42.x86_64.rpm.html#:~:text=Package%20version%20%093.14.2,Package%20architecture%20%09x86_64

### Is this bug present after an update or on a fresh install?

Updated from a minor version (ex. 3.4.2 to 3.4.4)

### Are you using the Nextcloud Server Encryption module?

Encryption is Enabled

### Are you using an external user-backend?

- [ ] Default internal user-backend
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Additional info

This is a *partial* recurrence of https://github.com/nextcloud/desktop/issues/6594#issue-2213769118 - a regression.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/nextcloud/desktop/issues/7419#issue-2614702441

This affects more than merely *toolbar* button iconography, actually:

> ![Image](https://github.com/user-attachments/assets/a804b6fd-75ae-47c5-b006-c55cffb37d69)

**Comment by the-nic** [comment]
Note that also the following icons are not shown in when pressing the details button:

![Image](https://github.com/user-attachments/assets/037a8965-5d40-419e-a4ae-a0aa42a031ce)

Is this because there are no white/black themed variants of the svg files referenced?

**Comment by RokeJulianLockhart** [comment]
> Is this because there are no white/black themed variants of the svg files referenced?

@the-nic, I'd be surprised, since in examples like https://invent.kde.org/frameworks/breeze-icons/-/merge_requests/410#note_1054423, the application should fall back to Breeze's colourful icons.

Perhaps it's another example of https://gitlab.freedesktop.org/xdg/xdg-specs/-/issues/132, although that's merely a guess.

**Comment by the-nic** [comment]
kinfo data:
```
Operating System: Arch Linux 
KDE Plasma Version: 6.2.2
KDE Frameworks Version: 6.7.0
Qt Version: 6.8.0
Kernel Version: 6.6.58-1-lts (64-bit)
Graphics Platform: Wayland
Processors: 16 × AMD Ryzen 7 5800X 8-Core Processor
Memory: 31.2 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 2070 SUPER/PCIe/SSE2
```

**Comment by BatmansGehilfe** [comment]
I experience the same behaviour as @RokeJulianLockhart and @the-nic here with a different distribution (Tumbleweed). Just to add some context.

kinfo data: 
`Operating System: openSUSE Tumbleweed 20241027
KDE Plasma Version: 6.2.2
KDE Frameworks Version: 6.7.0
Qt Version: 6.8.0
Kernel Version: 6.11.5-1-default (64-bit)
Graphics Platform: Wayland
Processors: 8 × Intel® Core™ i7-10510U CPU @ 1.80GHz
Memory: 15.4 GiB of RAM
Graphics Processor: Mesa Intel® UHD Graphics`

**Comment by working-name** [comment]
Same or very similar issue on windows. Might be something about dark mode.
![Image](https://github.com/user-attachments/assets/6fffdf19-3054-46ae-b893-cb0562290d46)

**Comment by RokeJulianLockhart** [comment]
> https://github.com/nextcloud/desktop/issues/7419#issuecomment-2466272561

@working-name, thanks for that. It discounts [the aforementioned](https://github.com/nextcloud/desktop/issues/7419#issuecomment-2443787297):

> Perhaps it's another example of [gitlab.freedesktop.org/xdg/xdg-specs/-/issues/132](https://gitlab.freedesktop.org/xdg/xdg-specs/-/issues/132), although that's merely a guess.

**Comment by BatmansGehilfe** [comment]
I don't think it is about dark mode: I experience the same behaviour in "light mode".

![Image](https://github.com/user-attachments/assets/ab373ed5-8c86-48d6-aff5-6c8e628217cf)

**Comment by RokeJulianLockhart** [comment]
> https://github.com/nextcloud/desktop/issues/7419#issuecomment-2467985643

@BatmansGehilfe, I do too - the icons don't even recolour:

![Image](https://github.com/user-attachments/assets/b5ee8e1b-7d26-4183-be71-c5c8873dd030)

**Comment by nilsding** [comment]
I looked a bit at this issue today -- for some strange reason the `icon.source` property on `AbstractButton`s doesn't work with `image://` URLs (used for [`QQuickImageProvider`](https://doc.qt.io/qt-6/qquickimageprovider.html)) on Linux.  It's really strange that it's working fine on macOS and Windows though 🤔

**Comment by RokeJulianLockhart** [comment]
> https://github.com/nextcloud/desktop/issues/7419#issuecomment-2685857864

@nilsding, if *that's* the problem, then that sounds like a `QTBUG`. I shan't be surprised if a relevant report exists at Jira, but I'm not proficient at searching it.

I'd say that the solution is as simple as just converting that URI to a path before it's passed to the provider, but the docs don't appear to demonstrate that that's permitted.

**Comment by nilsding** [comment]
after some more checks I think this is more an issue with the _Breeze_ QQuick style rather than something inside Qt itself -- in fact there's already a bug report on the KDE bugzilla about `icon.source` not being handled in a way that QML usually does: https://bugs.kde.org/show_bug.cgi?id=433433

The AppImage builds we provide use the _Fusion_ QQuick style, and it works fine there as well.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/nextcloud/desktop/issues/7419#issuecomment-2745794768

@nilsding, thanks. I've requested clarification at [`#c5`](https://bugs.kde.org/show_bug.cgi?id=433433#c5), since that does appear plausible. I can try forcing Fusion on the RPM to confirm, if you provide the relevant CLI parameters or configuration file content.

**Comment by RokeJulianLockhart** [comment]
> The triage assignee for the aforecited KDE bug ([`id=433433`](https://bugs.kde.org/show_bug.cgi?id=433433)) has tentatively confirmed that this appears to be a symptom of it. Consequently, I believe that I should close this due to it being an upstream fault, unless anyone can confirm this on any DE other than KDE Plasma.

In retrospect, I forgot https://github.com/nextcloud/desktop/issues/7419#issuecomment-2466272561. Can it be caused by KDE if it reproduces on Windows? I would presume not. I'll attempt to ascertain whether this might be better taken to Qt themselves.

**Comment by major-mayer** [comment]
> @RokeJulianLockhart the issue title change you did, is incorrect, the icons are missing elsewhere also (like dolphin right click nextcloud share menu), not only in home menu

I can confirm this

**Comment by RokeJulianLockhart** [comment]
> https://github.com/nextcloud/desktop/issues/7419#issuecomment-2912151292

@GameBurrow and @major-mayer, thanks. Added to the issue description.

**Comment by rsnyman** [comment]
I have this issue too, it's been bugging me for many months, and it seems to be across different versions too.

```
Operating System: Debian GNU/Linux 12
KDE Plasma Version: 6.3.4
KDE Frameworks Version: 6.13.0
Qt Version: 6.8.2
Kernel Version: 6.12.27-amd64 (64-bit)
Graphics Platform: Wayland
Processors: 8 × 11th Gen Intel® Core™ i5-1135G7 @ 2.40GHz
Memory: 31.1 GiB of RAM
Graphics Processor: Intel® Iris® Xe Graphics
```

**Comment by the-nic** [comment]
I found that reverting 599bbbed2a3a0489c0acfdce2c98c6fa89a9eef0 adds the more-apps icon back into header menu

**Comment by poltpolt** [comment]
still present in 3.16.4

**Comment by lolcabanon** [comment]
Still present in 3.16.6, see [my comment on another issue](https://github.com/nextcloud/desktop/issues/6729#issuecomment-3091502614) for details.

**Comment by BatmansGehilfe** [comment]
Still present in 3.17.0 (I'm still using openSUSE Tumbleweed, KDE Plasma 6.4.4)

**Comment by questionlp** [comment]
I'm using the Flatpak version and still seeing the issue with 3.17.1 on KDE Plasma 6.4.4 (Fedora 42)

> Nextcloud Desktop Client Version 3.17.1daily (KDE)

<img width="470" height="597" alt="Image" src="https://github.com/user-attachments/assets/38414bc0-4358-4ca8-9e75-95dbbcd44709" />

**Comment by Mayurifag** [comment]
In archlinux still persists. As a workaround one may use `nextcloud-client-appimage-daily` AUR package, but that is not a solution at all.

**Comment by mgallien** [comment]
unsure what the desktop client could do
I use `QT_QPA_PLATFORMTHEME=gtk3 ./nextcloud` as a workaround
my best guess is that this could be discussed with the KDE maintainers of the qml style
I am closing it as I think at least for some of the visual bugs, we have no way to fix them (the double down arrow for example)
I also acknowledge that this is a real issue (I am impacted)

**Comment by beedell-roke** [comment]
> https://github.com/nextcloud/desktop/issues/7419#issuecomment-3284994975

@mgallien, I've asked at [`discuss.kde.org/t/39648`][1]. If anyone here is willing to register for KDE's Discourse instance, a like would help to ensure it becomes visible amongst the deluge of daily posts.

~~If anyone has significant reason to believe that this *is* a bug with the QQuickStyle, please file a bug at [`bugs.kde.org/enter_bug.cgi?product=frameworks-qqc2-desktop-style`][2] (or similar, if that component is incorrect), and link to it here, or at the cited Discourse thread.~~

[2]: https://bugs.kde.org/enter_bug.cgi?product=frameworks-qqc2-desktop-style

[1]: https://discuss.kde.org/t/is-nextcloud-desktops-qml-flayouts-inability-to-render-icons-a-bug-with-breezes-qquickstyle/39648?u=rokejulianlockhart

**Comment by JSGRD22** [comment]
Since it does not seem this will be solved in the near future, forcing or defaulting to the GTK3 theme could be an appropriate temporary measure to avoid delivering a broken experience to users out of the box.

**Comment by beedell-roke** [comment]
@JSGRD22, I disagree. I can cope without icons, but they need to be replaced by textual fallbacks when unavailable. (It'd be an accessibility improvement.) [^1]

[^1]: [`bugzilla.mozilla.org/show_bug.cgi?id=1762473#c0`][1]

[1]: https://bugzilla.mozilla.org/show_bug.cgi?id=1762473#c0

**Comment by poltpolt** [comment]
I also disagree and I think a fallback should be a default icon included as qt resource for example, you can easily use opensource qt-kde breeze theme icons as fallback ones

**Comment by SamuXzX** [comment]
I know the issue is closed and has been moved to KDE Discuss instance. Maybe it's useful to document:
I use the Flatpak version.
```
Operating System: Fedora Linux 42
KDE Plasma Version: 6.4.4
KDE Frameworks Version: 6.17.0
Qt Version: 6.9.1
Kernel Version: 6.16.7-200.fc42.x86_64 (64-bit)
Graphics Platform: Wayland
Processors: 4 × Intel® Core™ i5-2400 CPU @ 3.10GHz
Memory: 16 GiB of RAM (15.4 GiB usable)
Graphics Processor: llvmpipe
```

**Comment by beedell-roke** [comment]
> the issue is closed and has been moved to KDE Discuss instance

@SamuXzX, the place for this now is [`bugs.kde.org/show_bug.cgi?id=509438#c0`][1]. [^1]

[1]: https://bugs.kde.org/show_bug.cgi?id=509438#c0

[^1]: [`discuss.kde.org/t/39648/5`][2]

[2]: https://discuss.kde.org/t/is-nextcloud-desktops-qml-flyouts-inability-to-render-icons-a-bug-with-breezes-qquickstyle/39648/5?u=rokejulianlockhart

> Maybe it's useful to document

If you're gonna provide environment information for a Flatpak package, the `flatpak info` output is also equally as important:

<blockquote>

~~~YAML
        ID: com.nextcloud.desktopclient.nextcloud
       Ref: app/com.nextcloud.desktopclient.nextcloud/x86_64/stable
      Arch: x86_64
    Branch: stable
   Version: 3.17.1
   License: GPL-2.0+
Collection: org.flathub.Stable
  Download: 130.5 MB
 Installed: 307.5 MB
   Runtime: org.kde.Platform/x86_64/6.9
       Sdk: org.kde.Sdk/x86_64/6.9

    Commit: 81e007e217033d0c023c18913c5f0657297aa1618ed1528ade27f81fe7d3d5d3
    Parent: d293d5f3c5916dec4595a3b1ad9f7a7a008684d33fe882a92d34193880616ae0
   Subject: Merge pull request #214 from flathub/update-master-5d4d703 (5916735864e1)
      Date: 2025-08-19 15:01:44 +0000
~~~

</blockquote>

---

> https://github.com/nextcloud/desktop/issues/7419#issuecomment-3286202048

@poltpolt, that could merely mask the problem, however.

**Comment by mgallien** [comment]
> > [#7419 (comment)](https://github.com/nextcloud/desktop/issues/7419#issuecomment-3284994975)
> 
> [@mgallien](https://github.com/mgallien), I've asked at [`discuss.kde.org/t/39648`](https://discuss.kde.org/t/is-nextcloud-desktops-qml-flayouts-inability-to-render-icons-a-bug-with-breezes-qquickstyle/39648?u=rokejulianlockhart). If anyone here is willing to register for KDE's Discourse instance, a like would help to ensure it becomes visible amongst the deluge of daily posts.
> 
> ~If anyone has significant reason to believe that this _is_ a bug with the QQuickStyle, please file a bug at [`bugs.kde.org/enter_bug.cgi?product=frameworks-qqc2-desktop-style`](https://bugs.kde.org/enter_bug.cgi?product=frameworks-qqc2-desktop-style) (or similar, if that component is incorrect), and link to it here, or at the cited Discourse thread.~

I am subscribed with my KDE developer account to the bug you created on KDE side. I will try to follow directly from there

**Comment by beedell-roke** [comment]
> https://github.com/nextcloud/desktop/issues/7419#issuecomment-3292451667

@mgallien, apologies for my irrelevant comments on it thus far. [^1] 😆

[^1]: [`id=509438#c3`][1]

[1]: https://bugs.kde.org/show_bug.cgi?id=509438#c3

**Comment by Nitrousoxide** [comment]
This might be related to this bug previously reported on the flatpak page for the nextcloud app.  I was using Gnome in my issue report then, though I'm no longer using the app so I can't confirm if it's still present.

https://github.com/flathub/com.nextcloud.desktopclient.nextcloud/issues/132

It may not be a KDE specific bug.

**Comment by the-nic** [comment]
With kde plasma 6.7 and setting `QT_QUICK_CONTROLS_STYLE=org.kde.union` fixes the issue on the KDE side.

**Comment by nilsding** [comment]
@the-nic Thanks for testing this with Union, this is some really good news!

could you share a screenshot of the tray window with Union enabled?

**Comment by the-nic** [comment]
> [@the-nic](https://github.com/the-nic) Thanks for testing this with Union, this is some really good news!
> 
> could you share a screenshot of the tray window with Union enabled?

@nilsding 

overview:
<img width="685" height="868" alt="Image" src="https://github.com/user-attachments/assets/bc3fe2c3-d139-4c6c-ac9f-f77898f86829" />

file details:

<img width="685" height="868" alt="Image" src="https://github.com/user-attachments/assets/3c2496ce-da3d-4340-9ae7-1ec47ca1531e" />

Just minor graphical glitches in the settings where the background is dark grey but should be light grey, but this could be a union issue 🤷 :

<img width="1114" height="417" alt="Image" src="https://github.com/user-attachments/assets/8ce0f3cf-6bba-47ae-9400-cad8c0968629" />



This looks weird when enabling limits:

<img width="1017" height="247" alt="Image" src="https://github.com/user-attachments/assets/e4f3e622-fac8-425d-90de-d0af062c1b66" />

**Comment by ppc52776** [comment]
> With kde plasma 6.7 and setting `QT_QUICK_CONTROLS_STYLE=org.kde.union` fixes the issue on the KDE side.

Thanks for this information. It works finally !!
